### DISCUSSION TITLE
Can we use the latest versions please?

### DISCUSSION THREAD

**Discussion by aprilmintacpineda** [codewars/codewars.com#2660] [General]
```
console.log([].includes);
```

throws error

```
TSError: ⨯ Unable to compile TypeScript
solution.ts (1,16): Property 'includes' does not exist on type 'undefined[]'. (2339)
```

**Comment by hobovsky** [comment]
No, unfortunately we cannot always use the latest version, for various reasons.

If you want to request a new version of a language, you can do so in [Codewars runner repository](https://github.com/codewars/runner/issues).

**Reply by aprilmintacpineda** [reply]
Okay, but `array.includes` is an ES6 feature, how come that's not yet fully supported?

**Reply by aprilmintacpineda** [reply]
I would think that at this point codewars should have full support for ES6 already, right?

**Reply by ejini6969** [reply]
please refer to  [list of TS katas to update](https://github.com/Codewars/content-issues/wiki/List-of-TypeScript-Kata-to-Update-to-3.x)

**Reply by ejini6969** [reply]
> I would think that at this point codewars should have full support for ES6 already, right?

It requires manual update on some katas due to failure in auto updating, feel free to contribute if you wish to see changes being made and thanks for your efforts if so!

**Reply by aprilmintacpineda** [reply]
I'm not even sure if I have the right skills to contribute to that 😅  I merely pointed this out as I've already encountered a few Katas with similar errors above not just with `.includes`.

**Reply by aprilmintacpineda** [reply]
So, if the Katas themselves are the cause of the update failure, then does that mean that on some Katas the `.includes` will work? Also, can't the authors of the Katas be informed that they have to update it? And what update exactly are we talking about here... I don't really have any idea of the underlying architecture.

**Reply by ejini6969** [reply]
> So, if the Katas themselves are the cause of the update failure, then does that mean that on some Katas the `.includes` will work? Also, can't the authors of the Katas be informed that they have to update it? And what update exactly are we talking about here... I don't really have any idea of the underlying architecture.

Yes, it's an issue. Autors are not always active, some have been inactive for few years...Feel free to come to Discord -> Codewars for more queries and information ! ~~

**Reply by kazk** [reply]
We can't always support the latest version, but we do support TypeScript 4.2 with ES2020 target. That wiki page lists kata that failed to update automatically, and mentions what usually needs to be fixed to update.  
You can contribute by forking a kata after completing and fixing any compile errors.
